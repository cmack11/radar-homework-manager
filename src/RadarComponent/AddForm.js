import React from 'react';
import TaskForm from './TaskForm.js'
import SubjectForm from './SubjectForm.js'


const TASK_FORM = 'TaskForm';
const SUBJECT_FORM = 'SubjectForm';

export class AddForm extends React.Component {

  constructor(props) {
    super();

    this.state = this.getDefaultState();
  }

  getDefaultState() {
    let defaultState = {form:SUBJECT_FORM};
    return defaultState;
  }

  switchForm() {
    if(this.state.form === TASK_FORM)
      this.setState({form:SUBJECT_FORM})
    else if(this.state.form === SUBJECT_FORM)
      this.setState({form:TASK_FORM})
  }

  render() {

    if(!this.props.show) return null;
    let form;

     if(this.state.form === TASK_FORM) 
      form = <TaskForm width={320} taskTypes={this.props.taskTypes} subjectNames={this.props.subjectNames} switchForm={this.switchForm.bind(this)} isEditForm={false}/>
    else if(this.state.form === SUBJECT_FORM)
      form = <SubjectForm width={320} taskTypes={this.props.taskTypes} switchForm={this.switchForm.bind(this)}/> 
    return (
      <div style={{position:'absolute',bottom:0,right:0}} id='subjectButton'>
      {form}
      </div>
    );
  }
}

export default AddForm
