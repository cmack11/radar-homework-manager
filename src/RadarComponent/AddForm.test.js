import React from 'react';
import AddForm from './AddForm.js';
import { shallow, mount, render } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

beforeAll(() => {
  Enzyme.configure({ adapter: new Adapter() })
  jest.useFakeTimers();
});

const line = {
  center:{x:10,y:15},
  length:100,
}

const setLineAngle = function() {

}

const taskTypes = ['Type #1','Type #2','Type #3'];
const subjectNames = ['Subject #1','Subject #2','Subject #3'];

describe('initialization', () => {

  it('should initialize the AddForm', () => {
    const wrapper = shallow(
      <AddForm show taskTypes={taskTypes} subjectNames={subjectNames}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should not show the AddForm by returning null', () => {
    const wrapper = shallow(
      <AddForm taskTypes={taskTypes} subjectNames={subjectNames}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
})

describe('initialization', () => {

  it('should switch the form and then switch back', () => {
    const wrapper = shallow(
      <AddForm show taskTypes={taskTypes} subjectNames={subjectNames}/>,
    );
    let addForm = wrapper.instance();
    let spy = jest.spyOn(addForm,'switchForm');

    let form1 = addForm.state.form;
    addForm.switchForm();
    addForm.render();
    expect(wrapper).toMatchSnapshot();
    expect(addForm.state.form).not.toEqual(form1);

    let form2 = addForm.state.form;
    addForm.switchForm();
    addForm.render();
    expect(addForm.state.form).not.toEqual(form2);
    expect(addForm.state.form).toEqual(form1);
    expect(spy).toBeCalledTimes(2);
    expect(wrapper).toMatchSnapshot();


  });

  it('should not alter the form', () => {
    const wrapper = shallow(
      <AddForm show taskTypes={taskTypes} subjectNames={subjectNames}/>,
    );
    let addForm = wrapper.instance();
    let spy = jest.spyOn(addForm,'switchForm');
    addForm.state.form = 'UNKOWN_FORM'

    let form1 = addForm.state.form;
    addForm.switchForm();
    addForm.render();

    expect(addForm.state.form).toEqual(form1);
    expect(spy).toBeCalled();
    expect(wrapper).toMatchSnapshot();


  });
})






