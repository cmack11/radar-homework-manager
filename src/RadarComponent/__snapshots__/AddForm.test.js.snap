// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialization should initialize the AddForm 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddForm
    show={true}
    subjectNames={
      Array [
        "Subject #1",
        "Subject #2",
        "Subject #3",
      ]
    }
    taskTypes={
      Array [
        "Type #1",
        "Type #2",
        "Type #3",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(SubjectForm)
        show={true}
        subjectNames={
          Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ]
        }
        switchForm={[Function]}
        taskTypes={
          Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ]
        }
        width={320}
      />,
      "id": "subjectButton",
      "style": Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "closeForm": undefined,
        "show": true,
        "subjectNames": Array [
          "Subject #1",
          "Subject #2",
          "Subject #3",
        ],
        "switchForm": [Function],
        "taskTypes": Array [
          "Type #1",
          "Type #2",
          "Type #3",
        ],
        "width": 320,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(SubjectForm)
          show={true}
          subjectNames={
            Array [
              "Subject #1",
              "Subject #2",
              "Subject #3",
            ]
          }
          switchForm={[Function]}
          taskTypes={
            Array [
              "Type #1",
              "Type #2",
              "Type #3",
            ]
          }
          width={320}
        />,
        "id": "subjectButton",
        "style": Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeForm": undefined,
          "show": true,
          "subjectNames": Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ],
          "switchForm": [Function],
          "taskTypes": Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ],
          "width": 320,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`initialization should not alter the form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddForm
    show={true}
    subjectNames={
      Array [
        "Subject #1",
        "Subject #2",
        "Subject #3",
      ]
    }
    taskTypes={
      Array [
        "Type #1",
        "Type #2",
        "Type #3",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(SubjectForm)
        show={true}
        subjectNames={
          Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ]
        }
        switchForm={[Function]}
        taskTypes={
          Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ]
        }
        width={320}
      />,
      "id": "subjectButton",
      "style": Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "closeForm": undefined,
        "show": true,
        "subjectNames": Array [
          "Subject #1",
          "Subject #2",
          "Subject #3",
        ],
        "switchForm": [Function],
        "taskTypes": Array [
          "Type #1",
          "Type #2",
          "Type #3",
        ],
        "width": 320,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(SubjectForm)
          show={true}
          subjectNames={
            Array [
              "Subject #1",
              "Subject #2",
              "Subject #3",
            ]
          }
          switchForm={[Function]}
          taskTypes={
            Array [
              "Type #1",
              "Type #2",
              "Type #3",
            ]
          }
          width={320}
        />,
        "id": "subjectButton",
        "style": Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeForm": undefined,
          "show": true,
          "subjectNames": Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ],
          "switchForm": [Function],
          "taskTypes": Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ],
          "width": 320,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`initialization should not show the AddForm by returning null 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddForm
    subjectNames={
      Array [
        "Subject #1",
        "Subject #2",
        "Subject #3",
      ]
    }
    taskTypes={
      Array [
        "Type #1",
        "Type #2",
        "Type #3",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`initialization should switch the form and then switch back 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddForm
    show={true}
    subjectNames={
      Array [
        "Subject #1",
        "Subject #2",
        "Subject #3",
      ]
    }
    taskTypes={
      Array [
        "Type #1",
        "Type #2",
        "Type #3",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(TaskForm)
        show={true}
        subjectNames={
          Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ]
        }
        switchForm={[Function]}
        taskTypes={
          Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ]
        }
        width={320}
      />,
      "id": "subjectButton",
      "style": Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "closeForm": undefined,
        "show": true,
        "subjectNames": Array [
          "Subject #1",
          "Subject #2",
          "Subject #3",
        ],
        "switchForm": [Function],
        "taskTypes": Array [
          "Type #1",
          "Type #2",
          "Type #3",
        ],
        "width": 320,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(TaskForm)
          show={true}
          subjectNames={
            Array [
              "Subject #1",
              "Subject #2",
              "Subject #3",
            ]
          }
          switchForm={[Function]}
          taskTypes={
            Array [
              "Type #1",
              "Type #2",
              "Type #3",
            ]
          }
          width={320}
        />,
        "id": "subjectButton",
        "style": Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeForm": undefined,
          "show": true,
          "subjectNames": Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ],
          "switchForm": [Function],
          "taskTypes": Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ],
          "width": 320,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`initialization should switch the form and then switch back 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddForm
    show={true}
    subjectNames={
      Array [
        "Subject #1",
        "Subject #2",
        "Subject #3",
      ]
    }
    taskTypes={
      Array [
        "Type #1",
        "Type #2",
        "Type #3",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(SubjectForm)
        show={true}
        subjectNames={
          Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ]
        }
        switchForm={[Function]}
        taskTypes={
          Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ]
        }
        width={320}
      />,
      "id": "subjectButton",
      "style": Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "closeForm": undefined,
        "show": true,
        "subjectNames": Array [
          "Subject #1",
          "Subject #2",
          "Subject #3",
        ],
        "switchForm": [Function],
        "taskTypes": Array [
          "Type #1",
          "Type #2",
          "Type #3",
        ],
        "width": 320,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(SubjectForm)
          show={true}
          subjectNames={
            Array [
              "Subject #1",
              "Subject #2",
              "Subject #3",
            ]
          }
          switchForm={[Function]}
          taskTypes={
            Array [
              "Type #1",
              "Type #2",
              "Type #3",
            ]
          }
          width={320}
        />,
        "id": "subjectButton",
        "style": Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeForm": undefined,
          "show": true,
          "subjectNames": Array [
            "Subject #1",
            "Subject #2",
            "Subject #3",
          ],
          "switchForm": [Function],
          "taskTypes": Array [
            "Type #1",
            "Type #2",
            "Type #3",
          ],
          "width": 320,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
